### PR TITLE
Refactoring of filemerger tests

### PR DESCRIPTION
**Pull request by oshadura**

### CODE REVIEW

**Comment by pcanal** [comment]
@phsft-bot build

**Comment by pcanal** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=On

**Comment by pcanal** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=ON

**Comment by oshadura** [comment]
@pcanal failure of test on mac1014/cxx17  is not connected with my changes.

**Comment by pcanal** [comment]
@oshadura I believe the failure on mac1014/cxx17 is a missing dependency (race condition) in the CMakeFiles.

**Comment by pcanal** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=ON

**Comment by oshadura** [comment]
@phsft-bot build!

**Comment by oshadura** [comment]
@phsft-bot build!

File: root/io/filemerger/CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,359 @@
...
+                  DEPENDS hsimple-filemergerfile)
+
+      ROOTTEST_ADD_TEST(simplex2-default-compr-level9
+                  PRECMD  ${ROOT_hadd_CMD} -f9 -a hsimple9x2.root hsimple.root
+                  COMMAND ${ROOT_root_CMD} -q -l -b "${CMAKE_CURRENT_SOURCE_DIR}/testSimpleFile.C(\"hsimple9x2.root\",50000,9,870903,10)"
+                  POSTCMD ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_BINARY_DIR}/hsimple9x2.root ${CMAKE_CURRENT_BINARY_DIR}/hsimple92.root
+                  DEPENDS simple-default-compr-level9-datageneration)
+
+      ROOTTEST_ADD_TEST(simplex2-zlib-compr-level9
+                  PRECMD  ${ROOT_hadd_CMD} -f109 -a hsimple109x2.root hsimple.root
```
Comment (pcanal) [inline root/io/filemerger/CMakeLists.txt]: This seems to use the same data file (name) as simple-zlib-compr-level9-datageneration

Comment (oshadura) [reply root/io/filemerger/CMakeLists.txt]: I am confused, but here I am using hadd -a option, right?

**Comment by oshadura** [comment]
> The file 'execTestMultiMerge.C' seems unused?

Yes, I divided it in two separate macros, so execTestMultiMerge.C could be actually removed.

File: root/io/filemerger/CMakeLists.txt (hunk 1 above)
Comment (pcanal) [reply root/io/filemerger/CMakeLists.txt]: I missed the '-a' :) [which means that we need to bring addition to it either by making the test name more explicit or adding a comment.]  thanks.

**Comment by pcanal** [comment]
> > The file 'execTestMultiMerge.C' seems unused?
> Yes, I divided it in two separate macros, so execTestMultiMerge.C could be actually removed.

Sorry, I got confused in the diff.  Indeed it is not used and is even already removed.

**Comment by pcanal** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=ON

**Pull request merged**